File: lumen-Front/src/api/channel.ts
/**
 * Channel REST API 客户端（T26: Session→Channel 迁移）
 */

const API_BASE_URL = 'http://127.0.0.1:8888';

export interface ChannelInfo {
  id: string;
  name: string;
  type: 'chat' | 'rpg' | 'board' | 'manage';
  description: string;
  order: number;
  group: 'base' | 'adventure' | 'free' | 'manage';
  created_at: string;
  updated_at: string;
}

export interface ChannelMessageInfo {
  id: number;
  role: string;
  content: string;
  channel_id: string;
  session_id: string;
  created_at: string;
}

export async function listChannels(): Promise<ChannelInfo[]> {
  const res = await fetch(`${API_BASE_URL}/channels`);
  if (!res.ok) throw new Error(`获取频道列表失败: ${res.status}`);
  return res.json();
}

export async function createChannel(
  name: string, type: string, description: string = '', group: string = 'base'
): Promise<{ channel_id: string }> {
  const res = await fetch(`${API_BASE_URL}/channels`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ name, type, description, group }),
  });
  if (!res.ok) {
    const err = await res.json().catch(() => ({ detail: res.statusText }));
    throw new Error(err.detail || `创建频道失败: ${res.status}`);
  }
  return res.json();
}

export async function deleteChannel(id: string): Promise<void> {
  const res = await fetch(`${API_BASE_URL}/channels/${id}`, { method: 'DELETE' });
  if (!res.ok) throw new Error(`删除频道失败: ${res.status}`);
}

export async function getChannelMessages(
  channelId: string, limit: number = 50, sinceId?: number
): Promise<ChannelMessageInfo[]> {
  let url = `${API_BASE_URL}/channels/${channelId}/messages?limit=${limit}`;
  if (sinceId) url += `&since_id=${sinceId}`;
  const res = await fetch(url);
  if (!res.ok) throw new Error(`获取频道消息失败: ${res.status}`);
  return res.json();
}

File: api/routes/channel.py
"""
Channel REST API（T26: Session→Channel 迁移）

提供频道的 CRUD 和消息查询，供前端频道列表和断线重连补拉使用。
"""

import logging
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel

from lumen.services.storage.history import (
    list_channels, create_channel, delete_channel,
    get_channel_messages,
)

logger = logging.getLogger(__name__)

router = APIRouter()


# ── 请求模型 ──

class CreateChannelRequest(BaseModel):
    name: str
    type: str = "chat"
    description: str = ""
    group: str = "base"


# ── 频道 CRUD ──

@router.get("/channels")
async def api_list_channels():
    """列出所有频道"""
    try:
        return list_channels()
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/channels")
async def api_create_channel(req: CreateChannelRequest):
    """创建新频道"""
    try:
        channel_id = create_channel(
            name=req.name, channel_type=req.type,
            description=req.description, group_name=req.group,
        )
        return {"channel_id": channel_id}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.delete("/channels/{channel_id}")
async def api_delete_channel(channel_id: str):
    """删除频道"""
    try:
        if not delete_channel(channel_id):
            raise HTTPException(status_code=404, detail="频道不存在")
        return {"success": True}
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/channels/{channel_id}/messages")
async def api_channel_messages(channel_id: str, limit: int = 50, since_id: int = 0):
    """获取频道消息（支持 since_id 断线补拉）"""
    try:
        return get_channel_messages(channel_id, limit=limit, since_id=since_id)
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
